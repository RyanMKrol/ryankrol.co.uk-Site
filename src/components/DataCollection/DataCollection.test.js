import React from 'react';
import fetch from 'node-fetch';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import flushPromises from '../../utils/TestUtils';

import {
  FULL_OUTPUT_MOCK,
  REDUCED_OUTPUT_MOCK,
  REDUCED_OUTPUT_MOCK_WITH_SORTER,
  EXAMPLE_LISTENS_MOCK_ONE,
  EXAMPLE_LISTENS_MOCK_TWO,
} from './__mocks__/DataCollectionMocks';

jest.mock('node-fetch');

it('renders DataCollection with full output', async () => {
  fetch.mockResolvedValueOnce({ json: () => [EXAMPLE_LISTENS_MOCK_ONE, EXAMPLE_LISTENS_MOCK_TWO] });
  const collection = shallow(FULL_OUTPUT_MOCK);

  await flushPromises();

  expect(toJson(collection)).toMatchSnapshot();
});

it('renders DataCollection with reduced output', async () => {
  fetch.mockResolvedValueOnce({ json: () => [EXAMPLE_LISTENS_MOCK_ONE, EXAMPLE_LISTENS_MOCK_TWO] });
  const collection = shallow(REDUCED_OUTPUT_MOCK);

  await flushPromises();

  expect(toJson(collection)).toMatchSnapshot();
});

it('renders DataCollection with reduced output using custom sorter', async () => {
  fetch.mockResolvedValueOnce({ json: () => [EXAMPLE_LISTENS_MOCK_ONE, EXAMPLE_LISTENS_MOCK_TWO] });
  const collection = shallow(REDUCED_OUTPUT_MOCK_WITH_SORTER);

  await flushPromises();

  expect(toJson(collection)).toMatchSnapshot();
});

it('DataCollection does not throw with full output', async () => {
  fetch.mockResolvedValueOnce({ json: () => [EXAMPLE_LISTENS_MOCK_ONE, EXAMPLE_LISTENS_MOCK_TWO] });
  expect(() => mount(FULL_OUTPUT_MOCK)).not.toThrow();
});

it('DataCollection does not throw with reduced output', async () => {
  fetch.mockResolvedValueOnce({ json: () => [EXAMPLE_LISTENS_MOCK_ONE, EXAMPLE_LISTENS_MOCK_TWO] });
  expect(() => mount(REDUCED_OUTPUT_MOCK)).not.toThrow();
});
