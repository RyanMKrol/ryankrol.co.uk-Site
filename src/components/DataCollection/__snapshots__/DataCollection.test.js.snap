// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders DataCollection with full output 1`] = `
<div>
  <div
    className="collection"
  >
    <ListensItemMini
      albumLink="test-albumLink-one"
      albumName="test-albumName-one"
      artist="test-artist-one"
      playcount={1}
      thumbnail="test-thumbnail-one"
    />
    <ListensItemMini
      albumLink="test-albumLink-two"
      albumName="test-albumName-two"
      artist="test-artist-two"
      playcount={2}
      thumbnail="test-thumbnail-two"
    />
  </div>
</div>
`;

exports[`renders DataCollection with reduced output 1`] = `
<div>
  <div
    className="collection"
  >
    <ListensItemFull
      albumLink="test-albumLink-one"
      albumName="test-albumName-one"
      artist="test-artist-one"
      playcount={1}
      thumbnail="test-thumbnail-one"
    />
  </div>
</div>
`;

exports[`renders DataCollection with reduced output using custom sorter 1`] = `
<div>
  <div
    className="collection"
  >
    <ListensItemFull
      albumLink="test-albumLink-two"
      albumName="test-albumName-two"
      artist="test-artist-two"
      playcount={2}
      thumbnail="test-thumbnail-two"
    />
    <ListensItemFull
      albumLink="test-albumLink-one"
      albumName="test-albumName-one"
      artist="test-artist-one"
      playcount={1}
      thumbnail="test-thumbnail-one"
    />
  </div>
</div>
`;
